src/router/index.ts

```diff
@@ -1,11 +1,28 @@
 import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
 
-import Index from "../views/Index.vue";
+import HomeLayout from "../layouts/HomeLayout.vue";
+import PageContents from "../components/PageContents.vue";
+import Login from "../components/Login.vue";
 
 const routes: Array<RouteRecordRaw> = [
   {
     path: "/",
-    component: Index,
+    component: HomeLayout,
+    children: [
+      {
+        //landing page
+        path: "",
+        component: PageContents,
+      },
+      {
+        path: "PageContents",
+        component: PageContents,
+      },
+      {
+        path: "Login",
+        component: Login,
+      }
+    ],
```

Comment:
Good idea, but the "component" part of the router should link to views, rather than components. You have some views already set up, for example you have a Login.vue **view** that already uses the Login.vue **component**. 

So for this, I'd create views for each of the pages, and those would be further composed of components. 

i.e. Create a PageContents view, and use your Home view & Login view as the children components.

An example can be found here [on the Quasar routing with layouts documentation](https://quasar.dev/layout/routing-with-layouts-and-pages)

Notice how their "pages" are being used as the child components. In our case, our "pages" are "views". 
```typescript
// src/router/routes.js

import User from 'layouts/User'
import Profile from 'pages/Profile'
import Posts from 'pages/Posts'

const routes = [
  {
    path: '/user',

    // we use /src/layouts/User component which is imported above
    component: User,

    // hey, it has children routes and User has <router-view> in it;
    // It is really a Layout then!
    children: [
      // Profile page
      {
        path: 'profile', // here it is, route /user/profile
        component: Profile // we reference /src/pages/Profile.vue imported above
      },

      // Posts page
      {
        path: 'posts', // here it is, route /user/posts
        component: Posts // we reference /src/pages/Posts.vue imported above
      }
    ]
  }
]

export default routes
```

Reply:
unmoved zaks move of page contents and created the view for page contents. also moved the q-page tags to the view files. should be good to merge